stdlib/mime/src/main/java/org/ballerinalang/mime/nativeimpl/MimeEntityBody.java

```diff
@@ -0,0 +1,216 @@
+import static org.ballerinalang.mime.util.MimeUtil.getNewMultipartDelimiter;
+
+/**
+ * Utilities related to MIME entity body.
+ *
+ * @since 1.1.0
+ */
+public class MimeEntityBody {
+    private static final Logger log = LoggerFactory.getLogger(MimeEntityBody.class);
+
+    public static Object getBodyParts(ObjectValue entityObj) {
+        ArrayValue partsArray;
+        try {
+            String baseType = HeaderUtil.getBaseType(entityObj);
+            if (baseType != null && (baseType.toLowerCase(Locale.getDefault()).startsWith(MULTIPART_AS_PRIMARY_TYPE) ||
+                    baseType.toLowerCase(Locale.getDefault()).startsWith(MESSAGE_AS_PRIMARY_TYPE))) {
+                //Get the body parts from entity's multipart data field, if they've been already been decoded
+                partsArray = EntityBodyHandler.getBodyPartArray(entityObj);
+                if (partsArray == null || partsArray.size() < 1) {
+                    Channel byteChannel = EntityBodyHandler.getByteChannel(entityObj);
+                    if (byteChannel != null) {
+                        EntityBodyHandler.decodeEntityBody(entityObj, byteChannel);
+                        //Check the body part availability for the second time, since the parts will be by this
+                        // time populated from bytechannel
+                        partsArray = EntityBodyHandler.getBodyPartArray(entityObj);
+                        //Set byte channel that belongs to parent entity to null, once the message body parts have
+                        // been decoded
+                        entityObj.addNativeData(ENTITY_BYTE_CHANNEL, null);
+                    }
+                }
+                return partsArray;
+            } else {
+                return MimeUtil.createError(PARSING_ENTITY_BODY_FAILED, "Entity body is not a type of " +
+                        "composite media type. Received content-type : " + baseType);
+            }
+        } catch (Throwable err) {
+            return MimeUtil.createError(PARSING_ENTITY_BODY_FAILED,
+                                        "Error occurred while extracting body parts from entity: " + getErrorMsg(err));
+        }
+    }
+
+    public static Object getBodyPartsAsChannel(ObjectValue entityObj) {
+        try {
+            String contentType = getContentTypeWithParameters(entityObj);
+            if (isMultipart(contentType)) {
+                String boundaryValue = HeaderUtil.extractBoundaryParameter(contentType);
+                String multipartDataBoundary = boundaryValue != null ? boundaryValue : getNewMultipartDelimiter();
+                ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+                MultipartDataSource multipartDataSource = new MultipartDataSource(entityObj, multipartDataBoundary);
+                multipartDataSource.serialize(outputStream);
+                EntityBodyChannel entityBodyChannel = new EntityBodyChannel(new ByteArrayInputStream(
```

Comment:
Don't we need to close these input/output streams?

Reply:
Only the outputstreams need closing.